src/templates/bash-template

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+# Copyright 2020 AstroLab Software
+# Author: Yash Datta
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set -e
+
+if [ $# -le 1 ]; then
+  echo "Usage: --config <config-file>"
+  exit 1
+fi
+
+VARS=$@
+
+# Parse config and rules file paths
+while test $# -gt 0
+do
+  case "$1" in
+    --config)
+      shift
+      CONF_FILE_PATH=$1;;
+  esac
+  shift
+done
+
+# Convert file paths to their basenames, since spark just needs base names to refer to these files
+CONF_FILE=$(basename $CONF_FILE_PATH)
+
+VARS=${VARS/$CONF_FILE_PATH/$CONF_FILE}
+
+lib_dir=$(cd "$(dirname ${BASH_SOURCE[0]})"/..; pwd)
+
+${{template_declares}}
+
+KRB5_CONF=""
+
+LOG_FILE=$lib_dir/run.out
+
+echo "Running spark-submit..."
+
+spark-submit \
+--master ${SPARK_MASTER} \
+--deploy-mode cluster \
+--driver-memory 8g \
+--driver-cores 4 \
+--num-executors 5 \
+--executor-memory 2g \
+--executor-cores 4 \
+--files $CONF_FILE_PATH \
+--conf spark.rdd.compress=true \
+--conf spark.executor.heartbeatInterval=120s \
```

Comment:
This clashes with the default `spark.network.timeout` value:
```
Exception in thread "main" java.lang.IllegalArgumentException: requirement failed: The value of 
spark.network.timeout=120000ms must be no less than the value of spark.executor.heartbeatInterval=120000ms.
```

I would remove `--conf spark.executor.heartbeatInterval=120s` unless you have a good reason to keep it.

Reply:
removed